### PR TITLE
feat(context): add result offload plugin

### PR DESCRIPTION
**Pull request by lizradway**
## Description

TypeScript port of [sdk-python#2162](https://github.com/strands-agents/sdk-python/pull/2162) and [sdk-python#2222](https://github.com/strands-agents/sdk-python/pull/2222). Adds a `ContextOffloader` plugin that proactively intercepts oversized tool results via `AfterToolCallEvent`, stores them in an external storage backend, and replaces the in-context result with a truncated preview plus per-block storage references.

**What it does:**
- Estimates token count of tool results using `model.countTokens()` (same approach as Python's `count_tokens`)
- When a result exceeds `maxResultTokens` (default: 2,500), offloads each content block to storage
- Replaces text/JSON blocks with a truncated preview and stored references
- Replaces image/video/document blocks with metadata placeholders and stored references
- Registers a `retrieve_offloaded_content` tool by default so the agent can fetch full content on demand (can be disabled via `includeRetrievalTool: false`)
- Retrieval tool returns native content blocks (ImageBlock, DocumentBlock) for binary content
- Storage references are actionable — FileStorage returns file paths, S3Storage returns `s3://` URIs — so agents can use existing tools (file_read, bash) directly
- Gracefully degrades — if storage fails, the original result is preserved

**Includes:**
- `Storage` interface with `InMemoryStorage`, `FileStorage`, and `S3Storage` implementations (mirrors all three Python backends)
- `ContextOffloader` plugin implementing the `Plugin` interface
- 39 unit tests covering offloading behavior, storage backends, retrieval tool, constructor validation, and edge cases

**Usage:**
```typescript
import { Agent } from '@strands-agents/sdk'
import { ContextOffloader, InMemoryStorage } from '@strands-agents/sdk/vended-plugins/context-offloader'

const agent = new Agent({
  model,
  plugins: [new ContextOffloader({ storage: new InMemoryStorage() })],
})
```

## Documentation PR

https://github.com/strands-agents/docs/pull/772 (Python docs — TS docs TBD)

## Type of Change

New feature

## Testing

How have you tested the change?

- [X] I ran `npm run check`
- 39 new unit tests (plugin behavior, InMemoryStorage, FileStorage, retrieval tool)
- All existing tests pass with no regressions
- TypeScript build compiles cleanly

## Checklist
- [X] I have read the CONTRIBUTING document
- [x] I have added any necessary tests that prove my fix is effective or my feature works
- [X] I have updated the documentation accordingly
- [X] I have added an appropriate example to the documentation to outline the feature, or no new docs are needed
- [x] My changes generate no new warnings
- [X] Any dependent changes have been merged and published

----

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: strands-ts/src/vended-plugins/context-offloader/storage.ts [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+    this._contentTypes[filename] = contentType
+    await this._saveMetadata(fs)
+
+    const filePath = path.join(this._artifactDir, filename)
+    await fs.writeFile(filePath, content)
+
+    return filePath
+  }
+
+  async retrieve(reference: string): Promise<{ content: Uint8Array; contentType: string }> {
```
[the remark this answers is not in the document]
Comment (lizradway) [reply strands-ts/src/vended-plugins/context-offloader/storage.ts:152]: `store()` returns `path.join(artifactDir, filename)`, which preserves whatever form the user configured. relative paths stays relative, absolute  paths stays absolute, etc. `retrieve()` uses `path.resolve` which handles both forms. The user controls the contract by choosing their artifact directory.

**Review by opieter-aws — DISMISSED** [review]

**Review by opieter-aws — APPROVED** [review]
